ui/src/components/Investor/Investor.tsx

```diff
@@ -103,24 +104,61 @@ const Investor: React.FC<Props> = ({ onLogout }) => {
 
                             { allExchanges.length > 0 ?
                                 allExchanges.map(exchange => {
-                                    return exchange.contractData.tokenPairs.map(tokenPair => {
+                                    const tokenPairs = exchange.contractData.tokenPairs.map(tokenPair => {
                                         const [ base, quote ] = unwrapDamlTuple(tokenPair).map(t => t.label.toLowerCase());
 
                                         return <Menu.Item
                                             as={NavLink}
+                                            exact
                                             to={{
                                                 pathname: `${url}/trade/${base}-${quote}`,
                                                 state: {
+                                                    isCleared: false,
                                                     exchange: exchange.contractData,
                                                     tokenPair: unwrapDamlTuple(tokenPair)
                                                 }
                                             }}
                                             className='sidemenu-item-normal'
-                                            key={exchange.contractId}
+                                            key={`${base}${quote}`}
                                         >
                                             <p><ExchangeIcon/>{base.toUpperCase()}/{quote.toUpperCase()}</p>
                                         </Menu.Item>
                                     })
+
+                                    const clearedMarkets = exchange.contractData.clearedMarkets.map(marketListing => {
+                                        const listing = unwrapDamlTuple(marketListing);
+                                        const tokenPair = typeof listing[0] !== 'string' && listing[0];
+                                        const defaultCCP = typeof listing[1] === 'string' && listing[1];
+
+                                        if (!tokenPair || !defaultCCP) {
+                                            throw new Error("Expected token pair and default CCP")
+                                        }
+
+                                        const [ base, quote ] = unwrapDamlTuple(tokenPair).map(t => t.label.toLowerCase());
+
+                                        console.log("Hmmmmm: ", base, quote, defaultCCP, tokenPair)
```

Comment:
Can probably take this out

Reply:
Oops yeah, thought I got all of these. Thanks for catching!

Reply:
Found a few more unnecessary console.logs too, removed them